src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -0,0 +1,485 @@
+  private CompletableFuture<CompositePurchaseOrder> createPOandPOLines(CompositePurchaseOrder compPO) {
+    CompletableFuture<CompositePurchaseOrder> future = new VertxCompletableFuture<>(ctx);
+    final WorkflowStatus finalStatus = compPO.getWorkflowStatus();
+
+    try {
+      // we should always create PO and PO lines in PENDING status and transition to OPEN only when it's all set
+      // (e.g. PO lines are created, Inventory is updated, etc.)
+      if (finalStatus == OPEN) {
+        compPO.setWorkflowStatus(PENDING);
+      }
+      JsonObject purchaseOrder = convertToPurchaseOrder(compPO);
+      httpClient.request(HttpMethod.POST, purchaseOrder, resourcesPath(PURCHASE_ORDER), okapiHeaders)
+        .thenApply(HelperUtils::verifyAndExtractBody)
+        .thenApply(poBody -> {
+          // At this point only id is important so using JsonObject
+          compPO.setId(poBody.getString(ID));
+          return compPO;
+        })
+        .thenCompose(this::createPoLines)
+        .thenAccept(lines -> {
+          compPO.setCompositePoLines(lines);
+          compPO.setAdjustment(HelperUtils.calculateAdjustment(lines));
+        })
+        .thenCompose(v -> {
+          if (finalStatus == OPEN) {
+            return openOrder(compPO);
+          }
+          return completedFuture(null);
+        })
+        .thenAccept(v -> future.complete(compPO))
+        .exceptionally(e -> {
+          future.completeExceptionally(e.getCause());
+          return null;
+        });
+    } catch (Exception e) {
+      logger.error("Exception calling POST /orders-storage/purchase-orders", e);
+      future.completeExceptionally(e);
+    }
+    return future;
+  }
+
+  private CompletableFuture<List<CompositePoLine>> createPoLines(CompositePurchaseOrder compPO) {
+    List<CompletableFuture<CompositePoLine>> futures =
+      compPO.getCompositePoLines()
+            .stream()
+            .map(compositePoLine -> orderLineHelper.createPoLine(compositePoLine, compPO))
+            .collect(Collectors.toList());
+    return HelperUtils.collectResultsOnSuccess(futures);
+  }
+
+  private CompletableFuture<Void> fetchCompositePoLines(CompositePurchaseOrder compPO) {
+    if (isEmpty(compPO.getCompositePoLines())) {
+      return getCompositePoLines(compPO.getId(), lang, httpClient, ctx, okapiHeaders, logger)
+                        .thenCompose(pols -> CompletableFuture.allOf(pols.stream().map(poline -> {
+                          compPO.getCompositePoLines().add(poline);
+                          return completedFuture(null);
+                        }).toArray(CompletableFuture[]::new)));
+    }
+    return completedFuture(null);
+  }
+
+  private void changePoLineReceiptStatuses(CompositePurchaseOrder compPO) {
+    compPO.getCompositePoLines().forEach(poLine -> {
+      if (poLine.getReceiptStatus() == CompositePoLine.ReceiptStatus.PENDING) {
+        poLine.setReceiptStatus(CompositePoLine.ReceiptStatus.AWAITING_RECEIPT);
+      }
+    });
+  }
+
```

Comment:
nitpick... extra blank line

Reply:
This is done automatically based on [.editorconfig](https://github.com/folio-org/mod-orders/blob/master/.editorconfig):
```
insert_final_newline = true
```

Reply:
Removed...